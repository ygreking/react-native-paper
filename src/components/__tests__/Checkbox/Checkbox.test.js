import * as React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '../../Checkbox/Checkbox';

it('renders checked Checkbox with onPress', () => {
  const tree = renderer
    .create(<Checkbox status="checked" onPress={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders unchecked Checkbox with onPress', () => {
  const tree = renderer
    .create(<Checkbox status="unchecked" onPress={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders indeterminate Checkbox', () => {
  const tree = renderer
    .create(<Checkbox status="indeterminate" onPress={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders checked Checkbox with color', () => {
  const tree = renderer
    .create(<Checkbox status="checked" color="red" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders unchecked Checkbox with color', () => {
  const tree = renderer
    .create(<Checkbox status="checked" color="red" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders indeterminate Checkbox with color', () => {
  const tree = renderer
    .create(<Checkbox status="indeterminate" color="red" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
