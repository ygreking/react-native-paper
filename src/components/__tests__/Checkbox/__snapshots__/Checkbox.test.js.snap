// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checked Checkbox with color 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 24,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders checked Checkbox with onPress 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        Array [
          Object {
            "color": "#03dac4",
            "fontSize": 24,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders indeterminate Checkbox 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        Array [
          Object {
            "color": "#03dac4",
            "fontSize": 24,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders indeterminate Checkbox with color 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 24,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders unchecked Checkbox with color 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 24,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders unchecked Checkbox with onPress 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 18,
        "padding": 6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        Array [
          Object {
            "color": "#03dac4",
            "fontSize": 24,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
